google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -1123,6 +1125,9 @@ public Builder setHost(String host) {
     /** Enables gRPC-GCP extension with the default settings. */
     public Builder enableGrpcGcpExtension() {
       this.grpcGcpExtensionEnabled = true;
+      if (!userSpecifiedNumChannels) { // If numChannels is not set through options
+        this.numChannels = 8; // default when grpc-gcp channel pool is enabled
+      }
       return this;
     }
```

Comment:
```suggestion
    public Builder enableGrpcGcpExtension() {
      return this(null);
    }
```
Have the null check for numChannels inside `public Builder enableGrpcGcpExtension(GcpManagedChannelOptions options) {`.

Reply:
Removed duplicate code of `enableGrpcGcpExtension` and added default `numChannels` check.